### ISSUE TITLE
Post 0.7 deprecations

### ISSUE THREAD

**Issue by diego-plan9** [Qiskit/qiskit-terra#1646]
### What is the expected enhancement?

This issue keeps track of the items that have been marked as deprecated in **0.7+**, in order to remove them before 0.8:
* [x] visualization: `plot_circuit()`, `latex_circuit_drawer()`, `generate_latex_source()`, `matplotlib_circuit_drawer()`; fallback; `style` kwargs dict options.
* [x] wrapper: `load_qasm_string()`, `load_qasm_file()`
* [x] transpiler: `transpile_dag()` `format` kwarg
* [x] result: several `Result` methods
* [x] tools.compiler: `compile()` and `execute()` `skip_transpiler` arg
* [x] mapper: `CouplingMap` `couplingdict` parameter
* [x] quantuminfo: `Pauli` `v`, `w`, `pauli_group()` `case` as integer
* [x] tools.qi: `state_fidelity()`
* [x] exceptions: `QISKitError`

**Comment by mtreinish** [comment]
I'm not sure we actually ever deprecated `QISKitError` we just setup the alias for backwards compat. If we do plan to remove it we should mark it as deprecated (a warning will be difficult to emit because it's the base exception). That being said it's not like it's a lot of debt to keep around (only a couple lines) I'm not sure if it's worth the large breaking change to remove just that.

**Comment by diego-plan9** [comment]
For tracking, related PRs: #1664, #1659, #1666

**Comment by diego-plan9** [comment]
Thanks @Max-Reuter for taking care of most of the deprecations! The final one is `QISKitError` - while we did not emit a warning for practical reasons indeed, I think we actually did conceptually deprecate it, as it is marked `not for use`. With that in mind, and seeing that it is the last stronghold where we use the old capitalization (started at #1382) and we embarked into more breaking changes in the past, I'll submit a PR.
